public/app/features/dashboard/dashgrid/DashboardLibrary/CommunityDashboardSection.tsx

```diff
@@ -0,0 +1,306 @@
+        pages: apiResponse.pages,
+        datasourceType: ds.type,
+      };
+    } catch (err) {
+      console.error('Error loading community dashboards', err);
+      throw err;
+    }
+  }, [datasourceUid, currentPage, debouncedSearchQuery]);
+
+  const styles = useStyles2(getStyles);
+
+  // Determine what to show in results area
+  const dashboards = Array.isArray(response?.dashboards) ? response.dashboards : [];
+  const totalPages = response?.pages || 1;
+  const showEmptyState = !loading && (!response?.dashboards || response.dashboards.length === 0);
+  const showError = !loading && error;
+
+  const onUseCommunityDashboard = async (dashboard: GnetDashboard) => {
+    if (response) {
+      DashboardLibraryInteractions.itemClicked({
+        contentKind: 'community_dashboard',
+        datasourceTypes: [response.datasourceType],
+        libraryItemId: String(dashboard.id),
+        libraryItemTitle: dashboard.name,
+        sourceEntryPoint: 'datasource_page',
+        eventLocation: 'suggested_dashboards_modal_community_tab',
+      });
+    }
+
+    try {
+      // Fetch full dashboard from Gcom, this is the JSON with __inputs
+      const fullDashboard = await fetchCommunityDashboard(dashboard.id);
+      const dashboardJson = fullDashboard.json;
+
+      // Parse datasource requirements from __inputs
+      const dsInputs: DataSourceInput[] = dashboardJson.__inputs?.filter(isDataSourceInput) || [];
+
+      // Parse constant inputs - these always need user review
+      const constantInputs = parseConstantInputs(dashboardJson.__inputs || []);
+
+      // Try auto-mapping datasources, considering we could come from "build dashhoard" there should be a datasource
+      // instance selected
+      const mappingResult = tryAutoMapDatasources(dsInputs, datasourceUid || '');
+
+      // Decide whether to show mapping form or navigate directly
+      // Show mapping form if: (a) there are unmapped datasources OR (b) there are constants
+      const needsMapping = !mappingResult.allMapped || constantInputs.length > 0;
+
+      if (!needsMapping) {
+        // No mapping needed - all datasources auto-mapped, no constants
+        navigateToTemplate(dashboard.name, dashboard.id, datasourceUid || '', mappingResult.mappings);
+      } else {
+        // Show mapping form for unmapped datasources and/or constants
+        onShowMapping({
+          dashboardName: dashboard.name,
+          dashboardJson,
+          unmappedInputs: mappingResult.unmappedInputs,
+          constantInputs,
+          existingMappings: mappingResult.mappings,
+          onInterpolateAndNavigate: (mappings) =>
+            navigateToTemplate(dashboard.name, dashboard.id, datasourceUid || '', mappings),
+        });
+      }
+    } catch (err) {
+      console.error('Error loading community dashboard:', err);
+      // TODO: Show error notification
+    }
+  };
+
+  return (
+    <Stack direction="column" gap={2} justifyContent="space-between" height="100%">
```

Comment:
this is changing the input width and also making the pagination component to be stacked in the middle
```suggestion
    <Stack direction="column" gap={2} justifyContent="space-between">
```

| With height 100% | Without height 100% |
|--------|--------|
| <img width="1190" height="175" alt="image" src="https://github.com/user-attachments/assets/8fca80c8-bc5a-4bc8-9b03-c70694c6a9a0" /> | <img width="1189" height="154" alt="image" src="https://github.com/user-attachments/assets/1238f6ae-8a17-4e77-99d3-ceb7aa482817" /> |
<img width="1194" height="538" alt="Screenshot 2025-11-05 at 7 01 24 PM" src="https://github.com/user-attachments/assets/b82cd4f3-bc02-47a9-94bc-15870f0351eb" /> | <img width="1210" height="573" alt="image" src="https://github.com/user-attachments/assets/5cea58aa-de02-4093-9f54-f04ad307fdc4" /> |